### PR TITLE
Backend improvements v2

### LINKED ISSUE #42: Amélioration du service de note
### Description

En tant que développeur je veux pouvoir accèder plus facilement aux notions de groupes, de conversations et de notes dans le service de notes.

### Definition of Done

Réimplémenter les endpoints suivants selon le nouveau modèle de note :
- Notes :
    - `ListNote`
    - `ExportNote`
- Implémentation des droits :  
    - `InsertBlock`
    - `UpdateBlock`
    - `DeleteBlock`
- Recommedation :
    -  `generateNoteTagsToModelNote`

### Estimation

Estimée à **4** JH.

### PR DESCRIPTION
**Pull request by gabriel654165**
#### Description

Small checks and fixes on some endpoints
Fixes #42 

#### Changelog

Remove the groups authorizations on blocks endpoints
Added rights authorization and validation on ListNotes
Put findFilters on repository.go in `find()` function :
- Excluded `blocks[]` and `keywords[]` from method ListNoteInternal
- Excluded `members[]` from ListGroupInternal

### CODE REVIEW

**Review by edouard-sn — APPROVED** [review]
ExportNote unit test implem would be cool but else everything's good

File: models/mongo/groups.go [hunk 1]
```diff
@@ -158,7 +159,10 @@ func (repo *groupsRepository) ListGroupsInternal(ctx context.Context, filter *mo
 		}})
 	}
 
-	err := repo.find(ctx, query, &groups, lo)
+	requieredFields := bson.D{{"members", 0}}
```
Comment (rojas-diego) [inline models/mongo/groups.go:162]:
```suggestion
	requieredFields := bson.D{{"members", 0}, {"inviteLinks", 0}, {"invites", 0}, {"conversations", 0}}
```

File: notes.go [hunk 2]
```diff
@@ -148,12 +148,18 @@ func (srv *notesAPI) ListNotes(ctx context.Context, req *notesv1.ListNotesReques
...
+		}
+	} else {
+		_, err = srv.groups.GetGroup(ctx, &models.OneGroupFilter{GroupID: req.GroupId}, token.AccountID)
+		if err != nil {
+			return nil, statusFromModelError(err)
+		}
 	}
 
-	notes, err := srv.notes.ListNotesInternal(ctx, &models.ManyNotesFilter{AuthorAccountID: req.AuthorAccountId}, &models.ListOptions{Limit: 20, Offset: 0})
+	notes, err := srv.notes.ListNotesInternal(ctx, &models.ManyNotesFilter{AuthorAccountID: req.AuthorAccountId}, &models.ListOptions{Limit: req.Limit, Offset: req.Offset})
```
Comment (rojas-diego) [inline notes.go:162]: You're not passing in the `req.GroupId`?

File: notes.go [hunk 3]
```diff
@@ -148,12 +148,18 @@ func (srv *notesAPI) ListNotes(ctx context.Context, req *notesv1.ListNotesReques
 	}
 
 	// Check user is part of the group
-	_, err = srv.groups.GetGroup(ctx, &models.OneGroupFilter{GroupID: req.GroupId}, token.AccountID)
-	if err != nil {
-		return nil, statusFromModelError(err)
+	if req.GroupId == "" {
+		if token.AccountID != req.AuthorAccountId {
+			return nil, status.Error(codes.PermissionDenied, "Could get note of another account")
```
Comment (rojas-diego) [inline notes.go:153]:
```suggestion
			return nil, status.Error(codes.PermissionDenied, "could get note of another account")
```

File: validators/notes.go [hunk 4]
```diff
@@ -39,9 +39,16 @@ func ValidateDeleteNoteRequest(req *notespb.DeleteNoteRequest) error {
...
 		validation.Field(&req.AuthorAccountId, validation.Required),
 	)
+	if err == nil {
+		return err
+	}
+	err = validation.ValidateStruct(req,
+		validation.Field(&req.GroupId, validation.Required),
+	)
+	return err
 }
```
Comment (rojas-diego) [inline validators/notes.go:52]: `req.AuthorAccountId` and `req.GroupId` are not required! At least one has to be set but you can have

```
ListNotes(authorAccountId, groupId)
ListNotes(groupId)
ListNotes(authorAccountId)
```

**Review by rojas-diego — CHANGES_REQUESTED** [review]
Thanks! Could you add the unit tests?

File: validators/notes.go (hunk 4 above)
Comment (gabriel654165) [reply validators/notes.go:52]: yes thats the point of this code,

**Pull request merged**